### DISCUSSION TITLE
Disable debug logs?

### DISCUSSION THREAD

**Discussion by 64-bitman** [ghostty-org/ghostty#3614] [Q&A] [answered]
What the title says. Why is this even the default? It's insane.

**Comment by mitchellh** [comment]
It isn't the default. Debug logs only show up in `Debug` builds. 

> It's insane.

Be nice, I do this for fun.

**Reply by 64-bitman** [reply]
Sorry for the poor choice of words. I meant reduce the amount of logs that ghostty outputs in the release build.

**Reply by mitchellh** [reply]
What logs in particular are noisy? Like what messages exactly.

**Reply by 64-bitman** [reply]
> What logs in particular are noisy? Like what messages exactly.

These logs. They look like info and warning logs, and they fill up my entire system journal
![image](https://github.com/user-attachments/assets/8bedb0c0-2146-41c8-bd7f-7cc8bab3e57f)

**Comment by rganesan** [comment]
I also notice verbose OpenGL logging on Linux Mint 22:

```
Making GLX context 0x35b20c60 current to drawable 65011721                            
Making GLX context 0x35061020 current to drawable 65011738
Flushing GLX buffers for drawable 65011738 (window: 65011716), frame sync: no
Making GLX context 0x35b20c60 current to drawable 65011721
Making GLX context 0x35061020 current to drawable 65011738
Flushing GLX buffers for drawable 65011738 (window: 65011716), frame sync: no
Making GLX context 0x35b20c60 current to drawable 65011721
Making GLX context 0x35061020 current to drawable 65011738
```

This logging is continuous as long as ghostty window is in focus even if I don't do anything (not even move the cursor).

**Reply by rganesan** [reply]
This seems to because OpenGL debug messages are enabled by setting GDK_DEBUG environment variable in src/apprt/gtk/App.zig. gl-disable-gles setting is needed but outputting OpenGL debugs by default seems wrong.

**Reply by mitchellh** [reply]
We do this by default so that people reporting GTK GL issues show us a lot of what we want to know by default. But I agree we should probably loosen that now and ask for it explicitly.

**Reply by rganesan** [reply]
Thanks for responding quickly. Maybe a config setting? I'll work on a patch if this sounds reasonable.

**Reply by nebkor** [reply]
I would dearly love to be able to disable this via config; for now, I'm keeping a patch that just doesn't set the GDK_DEBUG env var on top of `main`.

**Reply by aaroseif** [reply]
I also have verbose OpenGL logging on Ubuntu 24.04 which is flooding my /var/log/syslog
Does anyone know how to disable?

![image](https://github.com/user-attachments/assets/22374a82-d11a-4011-9d5e-9b8bcf3ebf0a)

**Comment by jcollie** [comment, marked as answer]
#4662 added a config entry to disable the OpenGL debug logs and will be in the next release.

**Comment by wesQ3** [comment]
I have this issue as well, journal just filling up with these debug logs.

```
Jan 24 13:32:21 x com.mitchellh.ghostty.desktop[509914]: Flushing GLX buffers for drawable 46152758 (window: 46152745), frame sync: no
Jan 24 13:32:22 x com.mitchellh.ghostty.desktop[509914]: Making GLX context 0x244f61c0 current to drawable 46137353
Jan 24 13:32:22 x com.mitchellh.ghostty.desktop[509914]: Making GLX context 0x266031b0 current to drawable 46152758
Jan 24 13:32:22 x com.mitchellh.ghostty.desktop[509914]: Flushing GLX buffers for drawable 46152758 (window: 46152745), frame sync: no
Jan 24 13:32:22 x com.mitchellh.ghostty.desktop[509914]: Making GLX context 0x244f61c0 current to drawable 46137353
Jan 24 13:32:22 x com.mitchellh.ghostty.desktop[509914]: Making GLX context 0x266031b0 current to drawable 46152758
```

I built off main fd8cacaa with `-Doptimize=ReleaseFast`. I tried setting the config value from #4662 but I get an error from config reload.
```
gtk-opengl-debug: unknown field
```

 Maybe I don't quite understand.

**Reply by wesQ3** [reply]
I had some sort of cached process. Even after rebuilding with 1.1.0 I still saw the debug logs so I started to suspect I wasn't running the version I thought I was. 

This is resolved as expected with the merged PR.
